src/main/java/Parser.java

```diff
@@ -0,0 +1,145 @@
+                if (input.contains("by")) {
+                    int byPos = input.indexOf("by") + 3;
+                    String time = input.substring(byPos, input.length());
+                    try {
+                        String parsedTime = formatDate(time);
+                        String listForm = input.substring(9, notePos - 1) + "by " + parsedTime;
+                        return new AddCommand(new Deadline(listForm));
+                    } catch (Exception e) {
+                        return new AddCommand(new Deadline(echo));
+                    }
+                } else {
+                    return new AddCommand(new Deadline(echo));
+                }
+            } else {
+                //this.addToList(new Deadline(input.substring(9, input.length())));
+                return new AddCommand(new Deadline(input.substring(9, input.length())));
+            }
+        }
+        else if (input.contains("event")) {
+            if(checkEmpty(input, "event")) {
+                throw new DukeException("Much error! You have to describe your mission!");
+            }
+            if(checkPlan(input)) {
+                throw new DukeException("Oh no, you do not have a planned timing for your mission!");
+            }
+            if(input.contains("/")) {
+                int notePos = input.indexOf("/") + 1;
+                String note = input.substring(notePos, input.length());
+                String echo = input.substring(6, notePos - 1) + " ------> " + note;
+                if(input.contains("on")) {
+                    int byPos = input.indexOf("on") + 3;
+                    String time = input.substring(byPos, input.length());
+                    try {
+                        String parsedTime = formatDate(time);
+                        String listForm = input.substring(6, notePos - 1) + "on " + parsedTime;
+                        return new AddCommand(new Event(listForm));
+                    }
+                    catch(Exception e) {
+                        return new AddCommand(new Event(echo));
+                    }
+                }
+                else {
+                    return new AddCommand(new Event(echo));
+                }
+            }
+            else {
+                return new AddCommand(new Event(input.substring(6)));
+            }
+        }
+        else {
+            throw new DukeException("Oops! There is no such keyword!");
+        }
+    }
+
+    private static Boolean checkEmpty(String input, String keyWord) {
+        int keywordLength = keyWord.length();
+        String remainingDescription = input.substring(keywordLength, input.length());
+        if (remainingDescription.length() == 0 ) {
+            return true;
+        }
+        else if (remainingDescription.length() > 1 && remainingDescription.charAt(1) == 32) {
+            return true;
+        }
+        else if (remainingDescription.length() <= 1) {
+            return true;
+        }
+        else {
```

Comment:
Perhaps you could consider writing the next if-else block immediately after the closing curly brace of the previous if-else block? I think this is a very minor issue still.